st3/lsp_utils/_client_handler/decorator.py

```diff
@@ -0,0 +1,77 @@
+from ..api_wrapper_interface import ApiWrapperInterface
+from .interface import ClientHandlerInterface
+from LSP.plugin.core.typing import Any, Callable, Iterable, List, Optional, Union
+import inspect
+
+__all__ = [
+    "notification_handler",
+    "request_handler",
+    "register_decorated_handlers",
+]
+
+# the first argument is always "self"
+T_HANDLER = Callable[[Any, Any], None]
+T_SERVER_EVENTS = Union[str, Iterable[str]]
+
+_HANDLER_MARKS = {
+    "notification": "__handle_notification_events",
+    "request": "__handle_request_events",
+}
+
+
+def notification_handler(server_events: T_SERVER_EVENTS) -> Callable[[T_HANDLER], T_HANDLER]:
```

Comment:
I don't understand how such a decorator can work since the argument passed to decoration is the original function, I believe, not whatever `server_events` is.